src/library/autodj/autodjprocessor.cpp

```diff
@@ -505,6 +505,17 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
 TrackPointer AutoDJProcessor::getNextTrackFromQueue() {
     // Get the track at the top of the playlist.
     while (true) {
+        // Incase we start off with < minimum tracks
+        bool randomQueueEnabled = (((m_pConfig->getValueString(
+                    ConfigKey("[Auto DJ]", "EnableRandomQueue")).toInt())) == 1);
+        int minAutoDJCrateTracks = m_pConfig->getValueString(
+                    ConfigKey(kConfigKey, "RandomQueueMinimumAllowed")).toInt();
+        int tracksToAdd = minAutoDJCrateTracks - m_pAutoDJTableModel->rowCount();
+        while (randomQueueEnabled && (tracksToAdd > 0)) {
```

Comment:
You should move the random track addition above the while(true) loop above. Just do be sure nothing bad can happen. In case off missing tracks eg. a big external HD is missing, it may take long to test each of the x k tracks that is stored on the HD.